flink-fs-tests/src/test/java/org/apache/flink/hdfstests/ContinuousFileProcessingTest.java

```diff
@@ -217,10 +222,11 @@ public void testFileReadingOperatorWithIngestionTime() throws Exception {
 			tester.setProcessingTime(nextTimestamp);
 
 			// send the next split to be read and wait until it is fully read, the +1 is for the watermark.
-			tester.processElement(new StreamRecord<>(
-				new TimestampedFileInputSplit(modTimes.get(split.getPath().getName()),
-					split.getSplitNumber(), split.getPath(), split.getStart(),
-					split.getLength(), split.getHostnames())));
+			RunnableWithException runnableWithException = () -> tester.processElement(new StreamRecord<>(
+					new TimestampedFileInputSplit(modTimes.get(split.getPath().getName()),
+							split.getSplitNumber(), split.getPath(), split.getStart(),
+							split.getLength(), split.getHostnames())));
+			tester.enqueueMail(runnableWithException);
```

Comment:
If I understood you correctly, it would introduce tight coupling between the test and the implementation details (`mailbox`).
And so it won't be compatible with the upcoming FLIP-27 implementation and the test would have to be rewritten again.